src/gt4py/next/iterator/type_inference.py

```diff
@@ -570,6 +570,14 @@ def _infer_shift_location_types(shift_args, offset_provider, constraints):
                     )
                     constraints.add((current_loc_out, Location(name=axis.origin_axis.value)))
                     current_loc_out = Location(name=axis.neighbor_axis.value)
+                elif axis.__class__.__name__ == "DummyConnectivity":
```

Comment:
Please make the `DummyConnectivity` in Icon4Py adhere to the `Connectivity` Protocol (defined in `gt4py.next.common`), remove the code path here and instead make the branch above check for `axis` implementing the Protocol, i.e. `isinstance(axis, common.Connectivity` (instead of the isinstance check for `gtx.NeighborTableOffsetProvider`, `gtx.StridedNeighborOffsetProvider`).